src/FSharp.Data.GraphQL.Server.Middlewares/TypeSystemExtensions.fs

```diff
@@ -1,7 +1,68 @@
 namespace FSharp.Data.GraphQL.Server.Middlewares
 
 open FSharp.Data.GraphQL.Types
+open FSharp.Data.GraphQL.Ast
 
+/// A function that checks if a directive should be used in the exection of a query, or changed to a new directive.
+type DirectiveChooser = Directive -> Directive option
+
+/// Basic operations on DirectiveChoosers.
+module DirectiveChooser =
```

Comment:
`DirectiveChooser` has a module to help composing choosers. We can easily build choosers based on other choosers, or diverse conditions.